python/ray/workflow/step_executor.py

```diff
@@ -35,6 +35,10 @@
 
 def get_step_executor(step_options: "WorkflowStepRuntimeOptions"):
     if step_options.step_type == StepType.FUNCTION:
+        # prevent automatic lineage reconstruction
+        # TODO(suquark): Raise an error if the original "max_retries"
+        #  is set and greater than 0.
+        step_options.ray_options["max_retries"] = 0
```

Comment:
Let's print a warning at least

Reply:
I update the PR and will use the `max_retries` of Ray task instead of the workflow itself.